Cython/Utility/StringTools.c

```diff
@@ -113,6 +113,14 @@ static int __Pyx_PyUnicodeBufferContainsUCS4_BMP(Py_UNICODE* buffer, Py_ssize_t
 #endif
 
 static CYTHON_INLINE int __Pyx_UnicodeContainsUCS4(PyObject* unicode, Py_UCS4 character) {
+#if CYTHON_COMPILING_IN_LIMITED_API
+    // Note that from Python 3.7, the indices of FindChar account for wraparound so no
+    // need to check the length
+    Py_ssize_t idx = PyUnicode_FindChar(unicode, character, 0, -1, 1);
+    if (idx == -1) return 0;  // not found
+    else if (unlikely(idx < 0)) return -1; // error
+    else return 1; // found
```

Comment:
The first two conditions seem surprising due to the overlap. According to the docs, the error case is only `-2`. Seems worth checking for that.
```suggestion
    if (unlikely(idx == -2)) return -1;
    // >= 0: found the index, == -1: not found
    return idx >= 0;
```